src/main/java/org/broadinstitute/hellbender/utils/diffengine/BAMDiffableReader.java

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.hellbender.utils.diffengine;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.io.*;
+import java.util.Arrays;
+
+/**
+ * Class implementing diffnode reader for BAM/SAM/CRAM files.
+ */
+final class BAMDiffableReader implements DiffableReader {
+
+    private static final byte[] BAM_MAGIC = "BAM\1".getBytes();
```

Comment:
Gross...this is a dupe of `BAMFileConstants.BAM_MAGIC` (which is package-protected, unfortunately). We should really patch htsjdk so that either the magic value is accessible or `SAMFileReader.isBAMFile()` is `public static`.